src/components/Body/index.tsx

```diff
@@ -58,20 +88,118 @@ function Game() {
     };
   }
 
+  function containsInWinningArrays(arr: number[][], games: number[]) {
+    return arr.find((a) => a.every((b) => games.includes(b)));
+  }
+
+  function setStyleRisk(games: number[]) {
+    const riskStyles = [
+      { array: horizontalArrays, style: horizontalArraysRiskStyle },
+      { array: verticalArrays, style: verticalArraysRiskStyle },
+      { array: leftDiagonalArrays, style: leftDiagonalArraysRiskStyle },
+      { array: rightDiagonalArrays, style: rightDiagonalArraysRiskStyle },
+    ];
+
+    riskStyles.forEach(({ array, style }) => {
+      if (containsInWinningArrays(array, games)) {
+        setRisk(style);
+      }
+    });
+  }
+
+  function restartGame(hasCountdown: boolean = true) {
+    if (hasCountdown) {
+      setCounter(5);
+    }
+    counter > 0 &&
+      setTimeout(() => {
+        setCounter((currentValue) => currentValue - 1);
+        if (counter <= 1) setCardList(initialCardListState);
+      }, 1000);
+  }
+
   useEffect(() => {
-    const jogos = getGames(cardList);
+    const games = getGames(cardList);
+    let xWin = containsInWinningArrays(potentialPlays, games.x);
+    let oWin = containsInWinningArrays(potentialPlays, games.o);
+    let tie = !xWin && !oWin && cardList.every((c) => c.letter !== "");
+
+    if (tie) {
+      setPointsTie((current) => current + 1);
+      restartGame();
+      return;
+    }
 
-    let xWin = potentialPlays.some((a) => a.every((b) => jogos.x.includes(b)));
-    let oWin = potentialPlays.some((a) => a.every((b) => jogos.o.includes(b)));
+    let winner: number[] = [];
+    if (xWin?.length) {
```

Comment:
Theres no need to check `if(xWin?.length)`, you can simply do `if(xWin)` because if it exist, it also has some length